### PR TITLE
Offload blocking operations in ECS to threadpool

### PR DESCRIPTION
**Pull request by FyzHsn**
- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation and CHANGELOG accordingly.
- [ ] I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by AlejandroEsquivel** [comment]
Hey @FyzHsn, I do think it might make sense to reduce some of the repetition by the introduction of something like:
```python
def _execute_partial_in_threadpool(partial_func):
  loop = asyncio.get_running_loop()
  future = await loop.run_in_executor(None, partial_object)
  return future
```

then can be used as such:

```python
run_task_response = _execute_partial_in_threadpool(partial(
            ecs.run_task,
            taskDefinition=self.ecs_task_family_name,
            launchType="FARGATE",
            cluster=self.ecs_cluster_name,
            count=1,
            networkConfiguration={
                "awsvpcConfiguration": {
                    "subnets": [self.ecs_task_subnet_id],
                    "securityGroups": [self.ecs_task_security_group_id],
                    # This is only needed if we're using public subnets
                    "assignPublicIp": "ENABLED",
                },
            },
))
     
```

This will also decouple which executor is used to execute the partials from all of the individual functions.

**Comment by FyzHsn** [comment]
> Hey @FyzHsn, I do think it might make sense to reduce some of the repetition by the introduction of something like:
> 
> ```python
> def _execute_partial_in_threadpool(partial_func):
>   loop = asyncio.get_running_loop()
>   future = await loop.run_in_executor(None, partial_object)
>   return future
> ```
> 
> then can be used as such:
> 
> ```python
> run_task_response = _execute_partial_in_threadpool(partial(
>             ecs.run_task,
>             taskDefinition=self.ecs_task_family_name,
>             launchType="FARGATE",
>             cluster=self.ecs_cluster_name,
>             count=1,
>             networkConfiguration={
>                 "awsvpcConfiguration": {
>                     "subnets": [self.ecs_task_subnet_id],
>                     "securityGroups": [self.ecs_task_security_group_id],
>                     # This is only needed if we're using public subnets
>                     "assignPublicIp": "ENABLED",
>                 },
>             },
> ))
>      
> ```
> 
> This will also decouple which executor is used to execute the partials from all of the individual functions.

Sweet! That's an awesome suggestion. Making the change now

File: covalent_ecs_plugin/ecs.py [hunk 1]
```diff
@@ -160,95 +159,111 @@ async def _upload_task(self, function, args, kwargs, task_metadata) -> None:
...
             s3.upload_file(function_file.name, self.s3_bucket_name, s3_object_filename)
 
-    async def submit_task(self, task_metadata: Dict, identity: Dict) -> str:
+    async def _upload_task(
+        self, function: Callable, args: List, kwargs: Dict, task_metadata: Dict
+    ):
+        """Wrapper to make boto3 s3 upload calls async."""
+        dispatch_id = task_metadata["dispatch_id"]
+        node_id = task_metadata["node_id"]
+        loop = asyncio.get_running_loop()
```
Comment (AlejandroEsquivel) [inline covalent_ecs_plugin/ecs.py:168]: should use `_execute_partial_in_threadpool` for consistency

File: covalent_ecs_plugin/ecs.py [hunk 2]
```diff
@@ -267,7 +282,8 @@ async def run(self, function: Callable, args: List, kwargs: Dict, task_metadata:
 
         await self._poll_task(task_arn)
 
-        return await self.query_result(task_metadata)
+        loop = asyncio.get_running_loop()
+        return await loop.run_in_executor(None, self.query_result, task_metadata)
```
Comment (AlejandroEsquivel) [inline covalent_ecs_plugin/ecs.py:286]: same here

**Review by AlejandroEsquivel — CHANGES_REQUESTED** [review]
Looks great @FyzHsn , just some very minor changes

**Review by AlejandroEsquivel — APPROVED** [review]

**Pull request merged**